lib/task/search/arPopulateTask.class.php

```diff
@@ -44,17 +44,33 @@ protected function configure()
 To exclude a document type, use the --exclude-types option. For example:
 
   php symfony search:populate --exclude-types="term,actor"
+
+To see a list of available document types that can be excluded use the --show-types option.
 EOF;
   }
 
   public function execute($arguments = array(), $options = array())
   {
     sfContext::createInstance($this->configuration);
 
+    // If show-types flag set, show types available to index
+    if (!empty($options['show-types']))
+    {
+      $this->log(sprintf('Available document types that can be excluded: %s', implode(', ', $this->availableDocumentTypes())));
+      $this->ask('Press the Enter key to continue indexing or CTRL-C to abort...');
+    }
+
     new sfDatabaseManager($this->configuration);
 
     $excludeTypes = (!empty($options['exclude-types'])) ? explode(',', strtolower($options['exclude-types'])) : null;
 
     QubitSearch::getInstance()->populate($excludeTypes);
   }
+
+  protected function availableDocumentTypes()
+  {
+    $types = array_keys(QubitSearch::getInstance()->loadMappings()->asArray());
+    sort($types);
+    return $types;
```

Comment:
Unfortunately PHP's sort is weird and sorts in place, returning a boolean (success or failure).